src/app/Library/CrudPanel/Traits/Relationships.php

```diff
@@ -44,6 +44,29 @@ public function getRelationInstance($field)
         return $relation_model->{$related_method}();
     }
 
+    /**
+     * Get the fields with specific relation types.
+     *
+     * @param string|array $relation_types
+     *
+     * @return array The fields with corresponding relation types.
+     */
+    public function getFieldsWithRelationType($relation_types): array
+    {
+        if (is_string($relation_types)) {
+            return collect($this->fields())
+            ->where('model')
+            ->where('relation_type', $relation_types)
+            ->toArray();
+        }
+        if (is_array($relation_types)) {
+            return collect($this->fields())
+            ->where('model')
+            ->whereIn('relation_type', $relation_types)
+            ->toArray();
+        }
```

Comment:
@pxpm we may be able to simplify this a little bit. Instead of doing things differently in case it's string or array, we can cast the string to array, and have only one set of instructions for the both. I believe something like this should work and be cleaner right?

```php
    public function getFieldsWithRelationType($relation_types): array
     {
          return collect($this->fields())
              ->where('model')
              ->whereIn('relation_type', (array)$relation_types)
              ->toArray();
    }
```

This would also solve another small problem I have with this function - that otherwise it leaves a scenario where nothing is returned. If the parameter is not a string and not an array, it would return nothing. This is highly unlikely of course, but... you know. Better cover all bases.

Reply:
Just refactored. I agree with you, I just added the `array casting` as a sperate var for readability of what is happening.